src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -0,0 +1,142 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GoPlaceholderCountInspection extends GoInspectionBase {
+  private static final Pattern PLACEHOLDER_PATTERN =
+    Pattern.compile("(%#?((\\[\\d+\\]|\\*?)?\\.?)*(\\d*\\.\\d*)*\\s?(v|T|t|b|c|d|o|q|x|X|U|b|e|E|f|F|g|G|s|q|x|X|p))");
+
+  private static final Pattern INDEXED_PLACEHOLDER_PATTERN = Pattern.compile("(?:\\[(\\d+)\\])");
+
+  // This holds the name of the known formatting functions and position of the string to be formatted
+  private static final Map<String, Integer> FORMATTING_FUNCTIONS = ContainerUtil.newHashMap(
+    Pair.pair("printf", 0),
+    Pair.pair("sprintf", 0),
+    Pair.pair("errorf", 0),
+    Pair.pair("fprintf", 1),
+    Pair.pair("fscanf", 1),
+    Pair.pair("scanf", 0),
+    Pair.pair("sscanf", 1),
+    Pair.pair("fatalf", 0),
+    Pair.pair("panicf", 0));
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitCallExpr(@NotNull GoCallExpr o) {
+        if (!isFormattingFunction(o)) return;
+
+        String name = getName(o);
+        Integer placeholderPosition = FORMATTING_FUNCTIONS.get(name);
+        List<GoExpression> arguments = o.getArgumentList().getExpressionList();
+        if (arguments.size() <= placeholderPosition) {
+          return;
+        }
+
+        int placeholdersCount = getPlaceholders(arguments.get(placeholderPosition));
+        int parameterCount = arguments.size() - placeholderPosition - 1;
+
+        if (placeholdersCount != parameterCount) {
+          String message = String.format("Got %d placeholder(s) for %d arguments(s)", placeholdersCount, parameterCount);
+
+          PsiElement element = o.getExpression();
+          if (placeholdersCount < parameterCount) {
+            element = arguments.get(placeholderPosition);
+          }
+          holder.registerProblem(element, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+        }
+      }
+    };
+  }
+
+  private static boolean isFormattingFunction(@NotNull GoCallExpr o) {
+    return FORMATTING_FUNCTIONS.containsKey(getName(o));
+  }
+
+  @Nullable
+  private static String getName(@NotNull GoCallExpr o) {
+    PsiReference psiReference = o.getExpression().getReference();
+
+    if (psiReference == null) {
+      return null;
+    }
+
+    PsiElement resolved = psiReference.resolve();
+
+    if (!(resolved instanceof GoFunctionOrMethodDeclaration)) {
+      return null;
+    }
+
+    resolved = ((GoFunctionOrMethodDeclaration)resolved).getIdentifier();
+    if (resolved == null) {
+      return null;
+    }
+
+    return resolved.getText().toLowerCase();
+  }
+
+  private static int getPlaceholders(@Nullable GoExpression argument) {
+    if (argument == null) {
+      return 0;
+    }
+
+    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(argument.getText());
```

Comment:
Here I bump yet again into the whole resolving to the final type issue that I have with other inspections as well (see https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1668 for example).

I'm not sure how to do it efficiently (not to mention that at some point I'll get stuck into how to do the resolving in case there's more than one step involved) and doing the whole resolving until the final type / value for each inspection makes me think that there has to be a better way to handle this (unless I don't understand how IntelliJ Platform / Java Runtime optimizations work to make this efficient).

(stupid question but) Would it be possible to have the GoType / Value attached to the nodes so that resolving / value retrieval doesn't happen every time?